### PR TITLE
Feature/state consistency

### PR DESCRIPTION
**Pull request by Maker-Mark**
### Issues

#53 
### Overview of Pull Request(PR)

Added necessary state updates on QuestionState when making selections to make basic judging on results page possible
### Technical Changes
- Step1: Approach selection is stored in QuestionState's "userApproach"
- Step2: Your code solution selection is stored in QuestionState's "userCodeBlocks"
- Step3: Time/Space complexity selection is stored in QuestionState's "userReasoning" (for now, will be migrated to separate time and space complexity)

### Visual Changes
- Added color updates for a selection
- Put the left bar on select cards for step one and two (need help with making it look good @Jackster001 :)
image

### Help!
@Jackster001 Please help me with that left tab and give an overall review of cosmetic changes
@ethen001 Please review technical/state changes

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "workbench.colorTheme": "Atom One Dark"
```
Comment (ethen001) [inline .vscode/settings.json]: Is This necesssary? or is it just theme preferences for your editor?

Comment (Maker-Mark) [reply .vscode/settings.json]: That's just my settings for my editor! I think it should be .gitignored, no?

Comment (ethen001) [reply .vscode/settings.json]: Yes, it should be in your .gitignore

File: src/questions/ComplexitySelection.js [hunk 2]
```diff
@@ -39,8 +45,8 @@ const styles = StyleSheet.create({
   cardStyle: {
     width: '12%',
     height: '50%',
-    marginLeft: 'auto',
-    marginRight: 'auto',
+    marginLeft: 20,
+    marginRight: 20,
```
Comment (ethen001) [inline src/questions/ComplexitySelection.js]: I think it is better if you use percentages. Screen sizes varies from phone to phone causing styling to not look the same

File: src/common/CodeBlock.js [hunk 3]
```diff
@@ -1,10 +1,53 @@
...
+const styles = StyleSheet.create({
+  codeItem: {
+    flexWrap: 'wrap',
+    flexShrink: 1,
+    flexDirection: 'row',
+    backgroundColor: '#f5f5f5',
+    borderRadius: 5,
+    marginLeft: 0,
+    marginRight: 'auto',
+    marginTop: 5,
```
Comment (ethen001) [inline src/common/CodeBlock.js]: Same Here. Please make it a percentage for consistency across phone screens

**Comment by Maker-Mark** [comment]
> Thank your for not only fixing this issue but refactoring all the code and making it cleaner. I suggested some small changes. Besides that, I think this looks very good

Absolutely!
Great points and suggestions, thank you.

File: .vscode/settings.json (hunk 1 above)
Comment (Maker-Mark) [reply .vscode/settings.json]: Added it to my .gitignore

**Pull request merged**